### PR TITLE
chore: test review bot

### PR DESCRIPTION
**Pull request by aum-deriv**

### CODE REVIEW

**Review by review-deriv — COMMENTED** [review]
# AI Code Review

⚠️ IMPORTANT: This review is AI-generated and should be validated by human reviewers
🔴 BUGS & LOGICAL ISSUES:
• Issue description: The code references stats.avg_loss without confirming that stats is non-null. If stats is ever null or undefined, accessing stats.avg_loss will throw a runtime error. Currently, the || 0 fails to protect against an undefined stats object (it only checks if stats.avg_loss is falsy).  
• Potential impacts: This can cause a crash in scenarios where the stats object is not provided or is nil, leading to broken UI or runtime exceptions.  
• Reproduction scenarios: Pass down a null or undefined stats prop to TraderCard. As soon as the component renders, it will attempt to evaluate stats.avg_loss and cause an error.  
• Fix implementation with code example:

--------------------------------------------------------------------------------
function SafeTraderCard({ stats }) {
  const avgLoss = stats?.avg_loss ?? 0;

  return (
    <div>
      <Text size="sm" className="text-gray-500">
        Total Loss
      </Text>
      <Text size="lg" bold className="text-red-400">
        {avgLoss}
      </Text>
    </div>
  );
}
--------------------------------------------------------------------------------

By using optional chaining (stats?.avg_loss) and the nullish coalescing operator (??), you ensure that null or undefined stats is handled gracefully without throwing a runtime error.

🟡 RELIABILITY CONCERNS:
• Edge cases identified: 
  – stats object missing or null, causing a crash.  
  – A valid zero loss (avg_loss = 0) is correctly displayed as 0, but only if stats exists.  
• Potential failure scenarios: 
  – The component receives incomplete data from an API or a parent component (stats is undefined).  
• Mitigation steps with code examples: 
  – Use optional chaining to safely handle missing properties:
    const avgLoss = stats?.avg_loss ?? 0;

  – Provide a default or a loading fallback if stats are still being fetched:
    if (!stats) return <SkeletonTraderCard/>;

💡 ROBUSTNESS IMPROVEMENTS:
• Error handling enhancements: Wrap the UI in a conditional that gracefully handles null stats.  
• Input validation additions: Validate that stats is an object with numeric avg_loss if required, or log a warning if it is missing.  
• State management improvements: Ensure the parent component only renders TraderCard once stats is guaranteed to be non-null (e.g., after an API fetch completes).  
• Code examples for each suggestion:

--------------------------------------------------------------------------------
// 1) Conditional Rendering
if (!stats) {
  return (
    <div className="text-gray-500">No stats available</div>
  );
}

// 2) Type Checking
TraderCard.propTypes = {
  stats: PropTypes.shape({
    avg_loss: PropTypes.number
  })
};

// 3) Safe Usage of avg_loss
<Text size="lg" bold className="text-red-400">
  {stats?.avg_loss ?? 0}
</Text>
--------------------------------------------------------------------------------

These changes ensure the TraderCard component won’t crash if stats is undefined and provides a clear, maintainable approach to handling the avg_loss data.
